### PR TITLE
[python][image] Save output.data images in OutputData format, not just strings

### PR DESCRIPTION
**Pull request by rossdanlm**
[python][image] Save output.data images in OutputData format, not just strings

This will make it easy for Ryan on frontend to know exactly what data type we need. This is also backwards compatible for existing AIConfigs that have data saved only as a text string

##test
Automated tests

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/608).
* __->__ #608
* #632

### CODE REVIEW

File: python/tests/parsers/test_dalle_parser.py [hunk 1]
```diff
@@ -14,11 +14,10 @@ async def test_serialize_basic(set_temporary_env_vars: None):
         "size": "1024x1024",
     }
     aiconfig = AIConfigRuntime.create()
-    serialized_prompts = await aiconfig.serialize(
+    serialized_prompt = await aiconfig.serialize(
```
Comment (saqadri) [inline python/tests/parsers/test_dalle_parser.py:17]: Doesn't `serialize` return an array?

File: python/src/aiconfig/default_parsers/dalle.py [hunk 2]
```diff
@@ -124,7 +138,7 @@ async def serialize(
                 model=model_metadata, parameters=parameters, **kwargs
             ),
         )
-        return [prompt]
+        return prompt
```
Comment (saqadri) [inline python/src/aiconfig/default_parsers/dalle.py:141]: Hmmmm, I think we should return an array always from `serialize`

**Review by saqadri — APPROVED** [review]
Looks good, just fix the return type for `serialize`

File: python/src/aiconfig/default_parsers/dalle.py (hunk 2 above)
Comment (rossdanlm) [reply python/src/aiconfig/default_parsers/dalle.py:141]: Thanks for clarifying, did diff here to fix issue: https://github.com/lastmile-ai/aiconfig/pull/632

**Pull request merged**